nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,503 @@
+      "org.jitsi.jicofo.BRIDGE_MUC" = "jvbbrewery@internal.${cfg.hostName}";
+    };
+
+    services.prosody = mkIf cfg.prosody.enable {
+      enable = mkDefault true;
+      modules = {
+        admin_adhoc = mkDefault false;
+        bosh = mkDefault true;
+        ping = mkDefault true;
+        roster = mkDefault true;
+        saslauth = mkDefault true;
+        tls = mkDefault true;
+      };
+      extraModules = [ "pubsub" ];
+      extraConfig = ''
+        certificates = "${config.services.prosody.dataDir}"
+
+        Component "conference.${cfg.hostName}" "muc"
+          storage = "memory"
+          -- modules: muc_meeting_id muc_domain_mapper
+
+          muc_room_locking = false
+          muc_room_default_public_jids = true
+
+        Component "internal.${cfg.hostName}" "muc"
+          storage = "memory"
+          -- muc_room_cache_size = 1000
+          admins = { "focus@auth.${cfg.hostName}", "jvb@auth.${cfg.hostName}" }
+
+        Component "focus.${cfg.hostName}"
+          component_secret = os.getenv("JICOFO_COMPONENT_SECRET")
+      '';
+      virtualHosts.${cfg.hostName} = {
+        enabled = true;
+        domain = cfg.hostName;
+        extraConfig = ''
+          authentication = "anonymous"
+          c2s_require_encryption = false
+          admins = { "focus@auth.${cfg.hostName}" }
+        '';
+      };
+      virtualHosts."auth.${cfg.hostName}" = {
+        enabled = true;
+        domain = "auth.${cfg.hostName}";
+        extraConfig = ''
+          authentication = "internal_plain"
+        '';
+      };
+    };
+    systemd.services.prosody.serviceConfig = mkIf cfg.prosody.enable {
+      EnvironmentFile = [ "/var/lib/jitsi-meet/secrets" ];
+      SupplementaryGroups = [ "jitsi-meet" ];
+    };
+
+    users.groups.jitsi-meet = {};
+    systemd.tmpfiles.rules = [
+      "d '/var/lib/jitsi-meet' 0750 root jitsi-meet - -"
+    ];
+
+    systemd.services.jitsi-meet-init-secrets = {
+      wantedBy = [ "multi-user.target" ];
+      before = [ "jicofo.service" "jitsi-videobridge2.service" ] ++ (optional cfg.prosody.enable "prosody.service");
+      serviceConfig = {
+        Type = "oneshot";
+      };
+
+      script = let
+        secrets = [ "VIDEOBRIDGE_SECRET" "JICOFO_COMPONENT_SECRET" "JICOFO_USER_SECRET" ];
+      in
+      ''
+        if [ ! -f /var/lib/jitsi-meet/secrets ]; then
+          ${concatMapStringsSep "\n" (var: ''
+            echo "${var}=$(tr -dc a-zA-Z0-9 </dev/urandom | head -c 64)" >> /var/lib/jitsi-meet/secrets
+          '') secrets}
+
+          chown root:jitsi-meet /var/lib/jitsi-meet/secrets
+          chmod 640 /var/lib/jitsi-meet/secrets
+        fi
+
+      ''
+      + optionalString cfg.prosody.enable ''
+        source /var/lib/jitsi-meet/secrets
+        ${config.services.prosody.package}/bin/prosodyctl register focus auth.${cfg.hostName} "$JICOFO_USER_SECRET"
+        ${config.services.prosody.package}/bin/prosodyctl register jvb auth.${cfg.hostName} "$VIDEOBRIDGE_SECRET"
```

Comment:
Does this properly update secrets if I change `/var/lib/jitsi-meet/secrets`?

Reply:
Yes, running `prosodyctl register` multiple times causes the password to be set to the one from the last invocation.